### ISSUE TITLE
Quering a field called variables results in duplicate definition

### ISSUE THREAD

**Issue by csoLs** [dotansimha/graphql-code-generator#1119]
**Describe the bug**
If you query for a field called `variables` the type definition generator will create two `Variables` type within the same namespace, see example from the live-demo:
<img width="1427" alt="screen shot 2019-01-04 at 15 39 25" src="https://user-images.githubusercontent.com/9143012/50693547-91fc7d80-1037-11e9-85ff-c40942d2b41a.png">

**To Reproduce**
Schema:
```
schema {
  query: Query
}
type Query {
  user(id: ID!): User
}
type User {
  id: ID!
  variables: UserVariables
}
type UserVariables {
  example: Boolean!
}
```

Document:
```
query findUser($userId: ID!) {
  user(id: $userId) {
    id
    variables {
      example
    }
  }
}
```
Config:
```
generates:
  client-types.ts:
    - typescript-common
    - typescript-client
```

**Expected behavior**
Should be able to generate even if we query for a field called `variables`.
Perhaps always prefix query variables type with the query name? I.e in my example above the output would be:
```
export namespace FindUser {
  export type FindUserVariables = {
    userId: string;
  }
...
  export type Variables = {
    __typename?: "UserVariables";
    example: boolean;
  }
}
```

**Comment by dotansimha** [comment]
Thanks @csoLs ! 
It's a general issue we have with the codegen, there is no code that handles a case of reserved words. 
As a workaround, I can suggest to alias this field and:
```
query findUser($userId: ID!) {
  user(id: $userId) {
    id
    userVariables: variables {
      example
    }
  }
}
```

**Comment by dotansimha** [comment]
Fixed in the recent refactor.

**Comment by dotansimha** [comment]
Fix available in 1.0.0 👍
